dandi/models.py

```diff
@@ -113,6 +113,104 @@ def diff_models(model1, model2):
             print(f"{field} is different")
 
 
+def to_datacite(dandiset):
+    from .metadata import migrate2newschema
+
+    meta = dandiset.metadata
+    newmeta = migrate2newschema(meta)
+
+    dandiset_id = dandiset.identifier
+
+    attributes = {}
+    attributes["identifiers"] = [
+        {"identifier": newmeta.doi, "identifierType": "DOI"},
+        # not sure if I can add dandi_id in the PDF documentation the only option is DOI...
+        {"identifier": dandiset_id, "identifierType": "Dandi"},
+    ]
+    attributes["titles"] = [{"title": newmeta.name}]
+    attributes["descriptions"] = [
+        # not sure what should be the type os use Other
+        # options are: Abstract, Methods, SeriesInformation,
+        # TableOfContents, TechnicalInfo, Other
+        {"description": newmeta.description, "descriptionType": "Other"}
+    ]
+    attributes["publisher"] = "DANDI Archive"
+    attributes["publicationYear"] = datetime.now().year
+    # not sure about it dandi-api had "resourceTypeGeneral": "NWB"
+    attributes["types"] = {"resourceType": "NWB", "resourceTypeGeneral": "Dataset"}
+    # newmeta has also attribute url, but it often empty
+    attributes["url"] = newmeta.url
+    # assuming that all licenses are from SPDX?
+    attributes["rightsList"] = [
+        {
+            "schemeURI": "https://spdx.org/licenses/",
+            "rightsIdentifierScheme": "SPDX",
+            "rightsIdentifier": el.name,
+        }
+        for el in newmeta.license
+    ]
+    # not sure if these is correct schema or should I provide newmeta.schemaVersion
+    # if not here, i'm not sure where newmeta.schemaVersion should go
+    attributes["schemaVersion"] = "http://datacite.org/schema/kernel-4"
+
+    contributors = []
+    creators = []
+    for contr_el in newmeta.contributor:
+        contr_dict = {
+            "name": contr_el.name,
+            "contributorName": contr_el.name,
+            "schemeURI": "orcid.org",
+            "affiliation": getattr(contr_el, "affiliation", []),
+        }
+        if isinstance(contr_el, Person):
+            contr_dict["nameType"] = "Personal"
+            if len(contr_el.name.split(",")) == 2:
+                contr_dict["familyName"], contr_dict["givenName"] = contr_el.name.split(
+                    ","
+                )
+        elif isinstance(contr_el, Organization):
+            contr_dict["nameType"] = "Organizational"
+
+        if RoleType("dandi:Author") in getattr(contr_el, "roleName"):
+            creators.append(contr_dict)
+        # should creators  be also contributors if they have more roles
+        # I don't see a way to add more roles to the Contributor
+        else:
+            contr_dict["contributorType"] = getattr(contr_el, "roleName")
```

Comment:
i would suggest adding a comma separated string.

Reply:
but the question is if the `creator` should be also a `contributor`. I do not have "contributorType" field for a `creator`, so if I add someone to a contributor list only, I don't see a way to keep other roles

Reply:
so any person who is an author is a creator and any creator who has other roles should be added as contributor.